galaxy/common/version.py

```diff
@@ -14,7 +14,7 @@
 #
 # You should have received a copy of the Apache License
 # along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
-
+from __future__ import unicode_literals
```

Comment:
This specific import is considered harmful, thus we don't use it in Core. 

however, you can use this snippet for py3fying some other bits:
```python
from __future__ import absolute_import, division, print_function
__metaclass__ = type
```

Reply:
I strongly disagree on using `__metaclass__ = type` magic. It implicitly changes behavior of python classes, while explicit inheritance from `object` class works both for Python 2 and 3.

Reply:
FWIW it makes every class in the module new-style. We even have a linter for ansible modules to require these two lines.

Reply:
Yep, but I am not aware of any other big python project that uses this approach. Unlike `__future__` imports that are standardized, this compatibility import looks more like a hack, since it is not widely used (project specific), implicit, global (on module level) and not required at all when using explicit inheritance from `object` class.
Considering all the above in my opinion it is a bad practice, which we are unlikely to follow.